nsq/conn.py

```diff
@@ -297,28 +286,21 @@ def send(self, data):
 
     def upgrade_to_tls(self, options=None):
         # in order to upgrade to TLS we need to *replace* the IOStream...
-        #
-        # first remove the event handler for the currently open socket
-        # so that when we add the socket to the new SSLIOStream below,
-        # it can re-add the appropriate event handlers. Default to TLSv1.2
-        # unless ssl_version is set otherwise.
-        self.io_loop.remove_handler(self.socket.fileno())
-
         opts = {
             'cert_reqs': ssl.CERT_REQUIRED,
-            'ca_certs': default_ca_certs(),
             'ssl_version': ssl.PROTOCOL_TLSv1_2
         }
         opts.update(options or {})
-        self.socket = ssl.wrap_socket(self.socket,
-                                      do_handshake_on_connect=False, **opts)
 
-        self.stream = tornado.iostream.SSLIOStream(self.socket, io_loop=self.io_loop)
-        self.stream.set_close_callback(self._socket_close)
+        fut = self.stream.start_tls(False, ssl_options=opts, server_hostname=self.host)
```

Comment:
I don't think we were checking if the TLS server certificate was valid for the hostname connected to. Now, if not `ssl.CERT_NONE`, this should check hostname ...